src/app/courses/view-courses/courses-view.component.ts

```diff
@@ -42,15 +42,24 @@ export class CoursesViewComponent implements OnInit, OnDestroy {
     private couchService: CouchService,
     private userService: UserService,
     private route: ActivatedRoute,
-    private coursesService: CoursesService
+    public coursesService: CoursesService
   ) { }
 
   ngOnInit() {
-    this.coursesService.courseUpdated$.pipe(takeUntil(this.onDestroy$)).subscribe(course => this.courseDetail = course);
-    this.route.paramMap.pipe(takeUntil(this.onDestroy$)).subscribe(
-      (params: ParamMap) => this.coursesService.requestCourse({ courseId: params.get('id'), forceLatest: true }),
-      error => console.log(error)
-    );
+    this.route.paramMap.pipe(takeUntil(this.onDestroy$))
+    .subscribe((params: ParamMap) => {
+      const courseId = params.get('id');
+      const getOpts: any = { courseIds: [ courseId ] };
+      if (this.parent) {
+        getOpts.opts = { domain: this.userService.getConfig().parentDomain };
+      }
+      this.coursesService.updateCourses(getOpts);
+      this.coursesService.requestCourse({ courseId: params.get('id'), forceLatest: true });
+    }, error => (error));
+    this.coursesService.courseUpdated$.pipe(takeUntil(this.onDestroy$))
```

Comment:
Please do not switch the order of these functions.  It's unnecessary and simply makes it harder to review.